src/amatrix.cc

```diff
@@ -141,6 +151,18 @@ inline AMatrix<T> AMatrix<T>::operator+(T v) const {
   return AMatrix<T>(matrix_ + v);
 }
 
+template <typename T>
+inline std::unique_ptr<AMatrix<T>> AMatrix<T>::operator+(const T* v) const {
```

Comment:
I think I see what you are doing here, but I think it will be somewhat confusing to have the operator have a very different memory behavior based on the pointer/reference usage.  It's also not great if we have to redefine matrix addition - if this operator is needed, can you do something like `plus_mat->matrix_ = plus_mat->matrix_ + *v` instead?

I wonder if redefining the default and copy constructors to use MmdVector backing with a #define/#ifdef or conditioning on a global flag would resolve this, and allow you to use the existing operators.

Reply:
Agreed that this variance in memory impl based on parameter is confusing, and something I would like to change also. I think an issue with a global var or an `#ifdef` approach is that some matrices are memory-mapped while most are not. So, it's not a system-wide on/off.

Perhaps inheritance could be a solution here?